### DISCUSSION TITLE
S1 Pro - F4 Board - Getting "Invalid Mesh" after every power cycle

### DISCUSSION THREAD

**Discussion by vash2695** [mriscoc/Ender3V2S1#714] [Q&A]
Recently this started happening 100% of the time. The printer will also occasionally fail to complete the leveling process with an M112 shutdown error.

Is this potentially hardware related?

**Comment by TobiasReich** [comment]
Sorry to hook into your comment but I have the Ender 3 S1 Pro F4 Board but assumed the firmware wouldn't support this printer since it has a touch display. Am I wrong?

**Reply by BogusF** [reply]
S1 an S1 pro have the same mainboard, so if you buy a e3v2 display (I did) you can install custom firmware no problem. Can't use it with the touchscreen though. There are even special configurations of the pro firmware to support 300°C hotend temp.

**Reply by vash2695** [reply]
As someone who switched screens for the upgrade it was totally worth it. I got really good results and loved the control I gained.

**Comment by mriscoc** [comment]
What mesh size are you trying?
Try with a 5x5, if it is larger you should disable HS mode.

**Reply by vash2695** [reply]
That actually may have been the case, as I usually do 6x6 or higher. The weird part was that this only happened intermittently. I had seen a reduction in print quality since it started as well. I was in the middle of a project so I had to move over to klipper to keep going, unfortunately, so I won't be able to test further. 

At this point I don't think it was related to anything inherent to the firmware, but rather a potential storage/processing error on the board (hence the move to klipper). Thank you for helping regardless!

**Reply by TobiasReich** [reply]
Sorry for hooking in again but would you still suggest the change? I'm thinking of purchasing one of the Ender 3 v2 displays if this really makes such a difference but it sounds like you eventually moved on.

**Reply by vash2695** [reply]
Yeas, absolutely. I did get great results with this firmware and any issues I had were either due to hardware issues or there not being a lot of flash space on the F4 board, but that's due to my own configs so no fault there. Klipper so far has also been excellent but installation was not nearly as easy as this.

**Reply by fuzelet** [reply]
Definitely worth it.  I have an S1 Pro and Ender 6 and replaced both of their screens with a S1 screen and have no problems. Creality is really holding back their machines by not open sourcing their touchscreens for better marlin support.
